engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceRejectionTest.java

```diff
@@ -90,4 +90,48 @@ public void shouldRejectCommandWhenAtLeastOneElementIdIsUnknown() {
             "Expected to activate element but no element found in process '%s' for element id(s): 'B', 'C'"
                 .formatted(PROCESS_ID));
   }
+
+  @Test
+  public void shouldRejectCommandWhenAtLeastOneElementIsInsideMultiInstance() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .userTask("A")
+                .subProcess(
+                    "subprocess",
+                    s -> s.embeddedSubProcess().startEvent().manualTask("B").manualTask("C").done())
+                .multiInstance(m -> m.zeebeInputCollectionExpression("[1,2,3]"))
+                .manualTask("task")
+                .endEvent()
+                .done())
+        .deploy();
+    final var processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+    RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withElementId("A")
+        .await();
+
+    // when
+    final var rejection =
+        ENGINE
+            .processInstance()
+            .withInstanceKey(processInstanceKey)
+            .modification()
+            .activateElement("B")
+            .activateElement("C")
+            .expectRejection()
+            .modify();
```

Comment:
❌  For completeness we should also activate an element that isn't inside a multi-instance.

The assertion below still covers that only the B and C elements are rejected.